test/async.c

```diff
@@ -41,7 +41,9 @@ static int blocking_getaddr(void *arg)
 
 	memset(&hints, 0, sizeof(hints));
 	hints.ai_family = AF_INET;
+#ifndef __ANDROID__
```

Comment:
Do we need to keep `AI_V4MAPPED` here for other targets or is it sufficient for this test to remove it entirely?

Reply:
could you please make this part into a separate PR ?